Extending the reach

laddbc opened the pull request:
TroyFletcher-

Thanks for creating synthwave. It mirrors the aesthetic in vogue when I started using emacs.

When I went to modify synthwave, I expanded the header to match that on other available themes. I do not believe I removed anything you had to say.

I needed more coverage for faces in Org mode and leuven-theme has some of the best faces coverage of any theme I have looked at. So I  started playing with bringing synthwave-theme over into something like leuven. I also added a group of customization variables to handle headline scaling in Org mode. 

I have been using it for almost two weeks and I have fixed up unreadable combinations I noted. Since it has been stable for a few days, I offer you my contribution in the form of a pull request.

Brian C. Ladd
laddbc

laddbc committed:
docs enhanced header comment cribbed from leuven-theme

Added GNU template header, mentioned leuven.

laddbc committed:
feat Added Org mode and colors

Modeled on leuven-theme.el, added named colors (synth-*, near-black,
etc.) and some face definitions. Grouped face definitions for the
theme.

Added (from leuven) everything that was org-*. Tweaked the theme over
a week of personal use. Faces used in prog-lang and org-mode seem
readable in most situations.

TroyFletcher commented on synthwave-theme.el:
```diff
@@ -1,139 +1,400 @@
+(let* (
+       (class '((class color) (min-colors 89)))
+       ;; Synthwave colors
+       (near-black "#100c0c")
```
I want a lighter background to better contrast the full black font-lock-comment-face. Original is "#181615" It can be assigned as near-black or at the default-background designation.

TroyFletcher commented on synthwave-theme.el:
```diff
@@ -1,139 +1,400 @@
+   `(underline ((,class (:underline t :background ,near-near-black :foreground ,near-white))))
+
+   ;; highlight faces
+   `(fringe ((,class (:underline t :background ,synth-orange :foreground ,synth-blue))))
```
I dislike the fringe color combination, I find the orange distracting. I would prefer a more muted tone that complements the modeline colorscheme. I see the value in an alternative background color, but want it to be more subtle. Foreground color may be aggressive or moderate.

TroyFletcher commented on synthwave-theme.el:
```diff
@@ -1,139 +1,400 @@
+       (grep-file-name `(:weight bold :foreground ,synth-cyan))
+       (grep-line-number `(:weight bold :foreground ,synth-green))
+
+       (highlight-current-line `(:background ,synth-subtle-cyan))
```
Prefer synth-deep-purple here.

TroyFletcher commented on synthwave-theme.el:
```diff
@@ -1,139 +1,400 @@
+   `(mode-line-inactive ((,class (:weight bold :background ,mode-line-inactive :foreground ,near-black))))
+
+   ;; font-lock faces
+   `(font-lock-builtin-face ((,class (:weight bold :foreground ,synth-pale-cyan))))
```
Lightening up the background washes out many of the colors. The background should be light, but keywords darker. Changing font-lock-builtin-face (and strings) to be darker helps. I'm testing with "#6cddf1" though I don't know I'd call it pale cyan, as it's bluer than the whole cyan palette. Perhaps swap with synth-lighter-blue

laddbc replied:
I honestly cannot see the difference between synth-light-cyan and "#6cddf1". I did NOT incorporate this change.

TroyFletcher reviewed (changes requested):
Sorry for the late reply, I missed this PR. Love the structure! It's light years ahead of where it was. Have some minor tweaks to maintain the lighter background and deeper colors to contrast the background. Less distracting fringe. Also less a fan of the orange/gold, but willing to try it out!

I've added some individual line comments.

Thanks again for the hard work!

laddbc committed:
fix: Applied most suggested changes from synthwave author

Lightened near-black
Changed fringe to synth-deep-cyan (from synth-gold)
Made highlight-current-line deep-purple

Did not: Change fontlock-builtin

laddbc commented:
Changed everything but the fontlock-builtin since I honestly could not see the difference. I may define a subtle-purple for highlighting the line; blue combinations with the deep-purple can be difficult for me to read.

TroyFletcher commented:
Great work!
